shtdlib.sh

```diff
@@ -178,6 +179,56 @@ function umask_decorator {
     return 1
 }
 
+# Bash behaviour option decorator
+# Allows changing/setting bash options for a function code block restoring
+# the original once execution of the function and it's calls is complete.
+# Requires an option name (see shopt) and a truthyness vale "true"/"false" or
+# other command that returns 0/1.
+# variables shopt_decorator_option_name and shopt_decorator_option_value
+# To use this add a line like the following (without #) as the first line of a function
+# shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return
+function shopt_decorator {
```

Comment:
Should the `decorator_option_value` and `decorator_option_name` variables be passed in as positional args instead?

Reply:
Ah ok that makes sense.